### ISSUE TITLE
Audit events not logged to audit logfile

### ISSUE THREAD

**Issue by inqueue** [elastic/elasticsearch#41318]
**Elasticsearch version** (`bin/elasticsearch --version`): Version: 7.0.0, Build: default/rpm/b7e28a7/2019-04-05T22:55:32.697037Z, JVM: 12

**OS version** (`uname -a` if on a Unix-like system): Linux json-sut-c1 3.10.0-957.5.1.el7.x86_64 `#1` SMP Fri Feb 1 14:54:57 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:
Audit log events are going to the server log instead of `<cluster_name>_audit.json`. 

**Steps to reproduce**:

 1. Set `xpack.security.audit.enabled: true` in elasticsearch.yml
 2. Inspect `<cluster_name>_audit.json`
 3. Inspect `<cluster_name>_server.json` (or `<cluster_name>.log`) and observe audit events there.

@jkakavas @ajoliveira

**Comment by albertzaharovits** [comment]
@inqueue the destination file of the audit appender is controlled from `config/log4j2.properties` .
I have tried the reproduction with a fresh 7.0 and it works **as expected** (one also needs `xpack.security.enabled: true` in elasticsearch.yml). The `config/log4j2.properties` has changed quite a lot to make logs JSON.

Can you please compare the `config/log4j2.properties` from your node with the stock reference?

**Comment by inqueue** [comment]
* MD5 checksums for default 7.0.0 `log4j2.properties` and my installation matched. 
* `xpack.security.enabled: true` was set (audit events were landing in `<cluster_name>.log` anyway)
* All works as expected after clean reinstallation of the coordinating node

The node having the issue was an upgrade from 5.6.15 -> 6.5 -> 6.7 -> 7.0.0. There were a few backups of `log4j2.properties` in the config directory which may have been in conflict. I spotted on another node in the same cluster where an `/etc/elasticsearch/x-pack/log4j2.properties` was left behind causing audit logging to land in `<cluster_name>_access.log`. Again, audit logging works as expected after cleaning it up.

Thanks for checking on it @albertzaharovits. Closing this until I find something specific related to upgrade.
